include/eosio/abi.hpp

```diff
@@ -320,4 +325,80 @@ abi_type* abi::add_type() {
    return add_type(*this, (T*)nullptr);
 }
 
+template <typename S>
+void to_json(const abi_extensions_type& ext_type, S& stream) {
+   stream.write('[');
+   bool first = true;
+   for(const auto& p : ext_type) {
+      if (first) {
+         first = false;
+      } else {
+         stream.write(',');
+      }
+      stream.write('{');
+      to_json(p.first, stream);
+      stream.write(':');
+      to_json(p.second, stream);
+      stream.write('}');
+   }
+   stream.write(']');
+}
+
+template <typename S>
+void to_json(const abi_def& def, S& stream) {
+   stream.write('{');
+   to_json("version", stream);
```

Comment:
Create a helper function to write each `name:value` pair to make the code more manageable